src/modules/users/users.service.ts

```diff
@@ -21,6 +43,13 @@ export class UsersService {
     return this.usersRepository.findOne(options);
   }
 
+  async findOneByEmail(email: string): Promise<User | null> {
+    const options: FindOneOptions<User> = {
+      where: { email },
+    };
+    return this.usersRepository.findOne(options);
```

Comment:
desde la linea 47-50 lo podés reemplazar por:  return await this.usersRepository.findOneBy({ email }); 
debería funcionar, directamente toma el email que recibe por parametro y hace la busqueda por ese valor.
De igual manera este método no deberia estar, ya que al momento de la creación se esta realizando la misma lógica.

Reply:
Si este lo hice por que lo necesitaba para hacer el custom decorator, pero al final lo hice de otra forma entonces quedo medio al vicio. Ahora lo cambiamos